compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -1650,29 +1657,38 @@ private BType checkListConstructorCompatibility(BType bType, BLangListConstructo
             case TypeTags.TYPEDESC:
                 // i.e typedesc t = [int, string]
                 listConstructor.isTypedescExpr = true;
-                List<BType> actualTypes = new ArrayList<>();
+                InferredTupleDetails inferredTupleDetails = new InferredTupleDetails();
                 for (BLangExpression expr : listConstructor.exprs) {
                     if (expr.getKind() == NodeKind.LIST_CONSTRUCTOR_SPREAD_OP) {
                         BLangExpression spreadOpExpr = ((BLangListConstructorSpreadOpExpr) expr).expr;
                         BType spreadOpExprType = checkExpr(spreadOpExpr, symTable.noType, data);
-                        actualTypes.addAll(getListConstSpreadOpMemberTypes(expr.pos, spreadOpExprType));
+                        updateInferredTupleDetailsFromSpreadMember(expr.pos, spreadOpExprType, inferredTupleDetails);
                         continue;
                     }
 
                     BType resultType = checkExpr(expr, symTable.noType, data);
+                    BType memberType;
```

Comment:
Shall we initialize with the `BType memberType = resultType` and remove the else part.